### PR TITLE
Network Troubleshooter dns check improve

### PR DESCRIPTION
**Pull request by yifguoMSFT**
Improve DNS check and other minor fixes

### CODE REVIEW

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js [hunk 1]
```diff
@@ -41,10 +41,48 @@ export var connectionFailureFlow = {
...
-    var ip = result.ip;
+async function runConnectivityCheck(hostname, port, dnsServers, diagProvider, lengthLimit) {
+    var subChecks = [];
+    var fallbackToPublicDns = false;
+    var nameResolvePromise = (async function checkNameResolve() {
+        var ip = null;
+        var subChecks = [];
+        for (var i = 0; i < dnsServers.length; ++i) {
+            var result = await diagProvider.nameResolveAsync(hostname, dnsServers[i]).catch(e => {
+                console.log(e);
```
Comment (XiaoxuHU) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js:52]: Remove console.log

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js [hunk 2]
```diff
@@ -41,10 +41,48 @@ export var connectionFailureFlow = {
...
+                }
+                break;
+            } else {
+                subChecks.push({ title: `Failed to resolve hostname '${hostname}' with ${dns}`, level: 1 });
+            }
+        }
+        return { ip, subChecks };
+    })();
+    var tcpPingPromise = diagProvider.tcpPingAsync(hostname, port, dnsServers.length).catch(e => {
+        console.log(e);
```
Comment (XiaoxuHU) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js:73]: Remove console.log

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js [hunk 3]
```diff
@@ -444,38 +501,51 @@ function checkNetworkConfigAndConnectivity(siteInfo, diagProvider, flowMgr, data
         kuduReachable = await kuduReachablePromise;
         if (kuduReachable) {
             var dnsCheckResult = await dnsCheckResultPromise;
-            dnsServer = dnsCheckResult.dnsServer;
+            dnsServers = dnsCheckResult.dnsServers;
```
Comment (XiaoxuHU) [inline AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js:504]: Maybe add a null check for `dnsCheckResult` before accessing its property

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js (hunk 1 above)
Comment (yifguoMSFT) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js]: fixed

File: AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js (hunk 2 above)
Comment (yifguoMSFT) [reply AngularApp/projects/app-service-diagnostics/src/app/shared/components/tools/network-checks/network-check-flows/connectionFailureFlow.js]: fixed

**Review by XiaoxuHU — APPROVED** [review]

**Pull request merged**